sunpy/io/jp2.py

```diff
@@ -71,6 +72,101 @@ def get_header(filepath):
 
     return [FileHeader(pydict)]
 
+def header_to_xml(header):
+    """
+    Converts image header metadata into an XML Tree that can be inserted into
+    a JP2 file header.
+
+    Parameters
+    ----------
+    header : `MetaDict`
+        A header dictionary.
+    """
+    # glymur uses lxml and will crash if trying to use
+    # python's builtin xml.etree
+    import lxml.etree as ET
+
+    fits = ET.Element("fits")
+
+    already_added = set()
+    for key in header:
+        # Some headers span multiple lines and get duplicated as keys
+        # header.get will appropriately return all data, so if we see
+        # a key again, we can assume it was already added to the xml tree.
+        if (key in already_added):
+            continue
+
+        # Add to the set so we don't duplicate entries
+        already_added.add(key)
+
+        el = ET.SubElement(fits, key)
+        data = header.get(key)
+        if type(data) == bool:
+            data = "1" if data else "0"
+        else:
+            data = str(data)
+
+        el.text = data
+
+    return fits
+
+def generate_jp2_xmlbox(header):
+    """
+    Generates the JP2 XML box to be inserted into the jp2 file.
+
+    Parameters
+    ----------
+    header : `MetaDict`
+        A header dictionary.
+    """
+    # glymur uses lxml and will crash if trying to use
+    # python's builtin xml.etree
+    import lxml.etree as ET
+    from glymur import jp2box
+
+    header_xml = header_to_xml(header)
+    meta = ET.Element("meta")
+    meta.append(header_xml)
+    tree = ET.ElementTree(meta)
+    return jp2box.XMLBox(xml=tree)
+
+def get_tmp_jp2_file_name(filename):
+    """
+    Returns a temporary file name for jp2 files since creating
+    a JP2 file with the correct header takes 2 passes.
+    """
+    return filename + ".tmp.jp2"
 
 def write(fname, data, header):
-    raise NotImplementedError("No jp2 writer is implemented.")
+    """
+    Take a data header pair and write a JP2 file.
+
+    Parameters
+    ----------
+    fname : `str`
+        File name, with extension.
+    data : `numpy.ndarray`
+        n-dimensional data array.
+    header : `dict`
+        A header dictionary.
+    """
+    import numpy as np
+    from glymur import Jp2k
```

Comment:
I moved numpy to the top but left glymur in the function. It's not the fastest import which is probably why it's also imported inside the read function already instead of the top of the file.